### PR TITLE
[IMP] iot: enable and diable ssh service on iotbox

### PR DESCRIPTION
**Pull request by viforget**
Before ssh was permanently activated, now there is a variable set to false by default to enable or disable it. The password generation depend of this variable too

3696215

### CODE REVIEW

File: addons/hw_drivers/tools/helpers.py [hunk 1]
```diff
@@ -409,6 +412,19 @@ def read_file_first_line(filename):
         with path.open('r') as f:
             return f.readline().strip('\n')
 
+def stop_ssh():
+    try:
+        subprocess.call(['sudo', 'systemctl', 'stop', 'ssh'])
+        subprocess.call(['sudo', 'systemctl', 'disable', 'ssh'])
+    except CalledProcessError as e:
```
Comment (Ysoroko) [inline addons/hw_drivers/tools/helpers.py:419]: except subprocess.CalledProcessError:

File: addons/hw_drivers/tools/helpers.py [hunk 2]
```diff
@@ -409,6 +412,19 @@ def read_file_first_line(filename):
...
+        _logger.error('An error encountered: %s', e)
+def start_ssh():
+    try:
+        subprocess.call(['sudo', 'systemctl', 'start', 'ssh'])
+        subprocess.call(['sudo', 'systemctl', 'enable', 'ssh'])
+    except CalledProcessError as e:
```
Comment (Ysoroko) [inline addons/hw_drivers/tools/helpers.py:425]: except subprocess.CalledProcessError:

File: addons/hw_drivers/tools/helpers.py [hunk 3]
```diff
@@ -409,6 +412,19 @@ def read_file_first_line(filename):
...
+        _logger.error('An error encountered: %s', e)
```
Comment (Ysoroko) [inline addons/hw_drivers/tools/helpers.py:426]: _logger.exception('Failed to start ssh')

File: addons/hw_drivers/tools/helpers.py [hunk 4]
```diff
@@ -409,6 +412,19 @@ def read_file_first_line(filename):
         with path.open('r') as f:
             return f.readline().strip('\n')
 
+def stop_ssh():
+    try:
+        subprocess.call(['sudo', 'systemctl', 'stop', 'ssh'])
+        subprocess.call(['sudo', 'systemctl', 'disable', 'ssh'])
+    except CalledProcessError as e:
+        _logger.error('An error encountered: %s', e)
```
Comment (Ysoroko) [inline addons/hw_drivers/tools/helpers.py:420]: _logger.exception('Failed to stop ssh')

File: addons/hw_drivers/tools/helpers.py [hunk 5]
```diff
@@ -198,15 +199,17 @@ def generate_password():
...
+        subprocess.call(('sudo', 'usermod', '-p', shadow_password, 'pi'))
 
-    with writable():
-        subprocess.call(('sudo', 'cp', '/etc/shadow', '/root_bypass_ramdisks/etc/shadow'))
+        with writable():
+            subprocess.call(('sudo', 'cp', '/etc/shadow', '/root_bypass_ramdisks/etc/shadow'))
 
-    return password
+        return password
+    return "ssh_service not activate"
```
Comment (Ysoroko) [inline addons/hw_drivers/tools/helpers.py:212]: Return a better description ("Activate ssh service in the db"?)

Comment (Ysoroko) [reply addons/hw_drivers/tools/helpers.py:212]: Enable ssh in the db first

Comment (qle-odoo) [reply addons/hw_drivers/tools/helpers.py:212]: maybe better "Enable ssh first in the db"

**Comment by viforget** [comment]
@odoo/rd-security is it possible to check this ?

**Comment by Ysoroko** [comment]
Not cool

**Pull request closed without merging**